### PR TITLE
BookingHrApproveHelper Junit

### PR DESCRIPTION
**Pull request by aashoo-rajput**

### CODE REVIEW

File: src/test/java/com/imagination/cbs/service/helper/BookingHrApproveHelperTest.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	private AdobeSignService adobeSignService;
+
+	@Mock
+	private BookingSaveHelper bookingSaveHelper;
+
+	@Mock
+	private EmailHelper emailHelper;
+	
+	@Test
+	public void shouldPrepareMailAndSendToHrWithBookingRevisionDetailsWhenUserIsAuthorized() throws IOException {
```
Comment (S-U-N-I-L) [inline src/test/java/com/imagination/cbs/service/helper/BookingHrApproveHelperTest.java:47]: can we break this test in multiple small test cases, having big test cases those test multiple things at one place is not recommended.

File: src/test/java/com/imagination/cbs/service/helper/BookingHrApproveHelperTest.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+		Booking booking = createBooking();
```
Comment (S-U-N-I-L) [inline src/test/java/com/imagination/cbs/service/helper/BookingHrApproveHelperTest.java:48]: use final for local ref.

File: src/test/java/com/imagination/cbs/service/helper/BookingHrApproveHelperTest.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+		when(bookingSaveHelper.getLatestRevision(booking)).thenReturn(bookingRevision);
+		when(html2PdfService.generateAgreementPdf(bookingRevision)).thenReturn(byteArrayOutputStream);
+		when(adobeSignService.uploadAndCreateAgreement((Mockito.any(InputStream.class)),Mockito.eq("service.pdf"))).thenReturn("CBS-111");
+		when(adobeSignService.sendAgreement("CBS-111", bookingRevision)).thenReturn("C-546");
+		when(loggedInUserService.getLoggedInUserDetails()).thenReturn(cbsUser);
+		when(bookingSaveHelper.saveBooking(booking, bookingRevision, 1006L,cbsUser)).thenReturn(booking);
+		
+		bookingHrApproveHelper.hrApprove(booking);
+		
+		verify(loggedInUserService).isCurrentUserInHRRole();
```
Comment (S-U-N-I-L) [inline src/test/java/com/imagination/cbs/service/helper/BookingHrApproveHelperTest.java:65]: these all should be tested in separate test cases.

File: src/test/java/com/imagination/cbs/service/helper/BookingHrApproveHelperTest.java [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+		ApprovalStatusDm approvalStatus = createApprovalStatusDm();
+		approvalStatus.setApprovalStatusId(1111L);
+		Booking booking = createBooking();
+		booking.setApprovalStatus(approvalStatus);
+		
+		when(loggedInUserService.isCurrentUserInHRRole()).thenReturn(true);
+		
+		bookingHrApproveHelper.hrApprove(booking);
+	}
+	private Booking createBooking() {
```
Comment (S-U-N-I-L) [inline src/test/java/com/imagination/cbs/service/helper/BookingHrApproveHelperTest.java:95]: can we move this method to some common place, from where we can use this domain creation in other test cases too.

File: src/test/java/com/imagination/cbs/service/helper/BookingHrApproveHelperTest.java (hunk 1 above)
Comment (aashoo-rajput) [reply src/test/java/com/imagination/cbs/service/helper/BookingHrApproveHelperTest.java:47]: Only one condition is included in this test that Authorized user can send email to HR.. and for UnAuthorized there is another test case, so not possible to break this test

File: src/test/java/com/imagination/cbs/service/helper/BookingHrApproveHelperTest.java (hunk 2 above)
Comment (aashoo-rajput) [reply src/test/java/com/imagination/cbs/service/helper/BookingHrApproveHelperTest.java:48]: Ok.. i will change it .

File: src/test/java/com/imagination/cbs/service/helper/BookingHrApproveHelperTest.java (hunk 3 above)
Comment (aashoo-rajput) [reply src/test/java/com/imagination/cbs/service/helper/BookingHrApproveHelperTest.java:65]: This is the part of one condition, cant be tested in seperate test cases.

Comment (S-U-N-I-L) [reply src/test/java/com/imagination/cbs/service/helper/BookingHrApproveHelperTest.java:65]: what i mean here by separate test case is to put every verify in different test, i m not sure why its not possible ? if you feel trouble then ping me .. i will help on this.

Comment (S-U-N-I-L) [reply src/test/java/com/imagination/cbs/service/helper/BookingHrApproveHelperTest.java:65]: all your when code we moved to a common or before method, so it wont be duplicate in every test case. we are trying to test the code in units ... thats whey our test cases should be small and oriented.

Comment (aashoo-rajput) [reply src/test/java/com/imagination/cbs/service/helper/BookingHrApproveHelperTest.java:65]: sure, tomorrow i will ping you and we discuss on this.

**Pull request merged**